src/transactions/TransactionFrame.cpp

```diff
@@ -299,18 +281,17 @@ TransactionFrame::getFee(LedgerHeader const& header,
                                   ProtocolVersion::V_11) ||
         !applying)
     {
-        int64_t feeBid = getInclusionFee();
-        int64_t flatFee = getFullFee() - feeBid;
         int64_t adjustedFee =
             *baseFee * std::max<int64_t>(1, getNumOperations());
 
         if (applying)
         {
-            return flatFee + std::min<int64_t>(getInclusionFee(), adjustedFee);
+            return declaredSorobanResourceFee() +
```

Comment:
I'm not too familiar with the surge pricing implementation, but how does that get taken into account? If we're in surge pricing, would the baseFee be the lowest inclusion fee in the txset?

Reply:
Yes, `baseFee` here is passed from external callers and is either 100 or whatever is the lowest fee for the tx set (in case of surge pricing).